src/java/picard/analysis/GcBiasMetricsCollector.java

```diff
@@ -125,13 +126,12 @@ public void acceptRecord(final GcBiasCollectorArgs args) {
             final SAMRecord rec = args.getRec();
             final String type;
             if (!rec.getReadUnmappedFlag()) {
-                final ReferenceSequence ref = args.getRef();
-                final byte[] refBases = ref.getBases();
-                StringUtil.toUpperCase(refBases);
-                final int refLength = refBases.length;
-                final int lastWindowStart = refLength - scanWindowSize;
-
-                if(referenceIndex != rec.getReferenceIndex() || gc == null){
+                if(referenceIndex != rec.getReferenceIndex() || gc == null || refBases == null){
```

Comment:
Why did you make refBases a class non-local? I think it could remain local, as you only need it here for the calculation of the gc (which should remain non-local)..